lib/features/popup-menu/PopupMenuComponent.js

```diff
@@ -81,18 +84,17 @@ export default function PopupMenuComponent(props) {
         return false;
       }
 
-      const search = [
+      const searchableFields = [
         entry.description || '',
         entry.label || '',
         entry.search || ''
-      ]
-        .join('---')
-        .toLowerCase();
+      ].map(string => string.toLowerCase());
 
+      // every word of `value` should be included in one of the searchable fields
       return value
         .toLowerCase()
         .split(/\s/g)
-        .every(term => search.includes(term));
+        .every(word => searchableFields.some(field => field.includes(word)));
```

Comment:
Could make sense to test cover this.

Reply:
Previously, we matched non-existent strings:

![brave_BG1L16Jg0d](https://github.com/bpmn-io/diagram-js/assets/7633572/78d1ee89-e777-4934-b1fd-89aec7d0c925)

This commit fixed it: https://github.com/bpmn-io/diagram-js/pull/847/commits/7c67215376ba2cb1e1430e8a67aca9358df9c5a4

I added tests as well: https://github.com/bpmn-io/diagram-js/pull/847/commits/baa739194463edd919203fb6dfeb0987ebae0aab

Reply:
Thanks. Fine with me, though the intention was to build the most simplest version possible (and assume no user would type `---` in the search box) :see_no_evil:.

Reply:
Well, I did. 🤡